src/App.js

```diff
@@ -1,37 +1,40 @@
 import React, { Component } from 'react';
 import './App.scss';
-import { FilmsList } from './components/FilmsList';
-import { NewFilm } from './components/NewFilm';
-import { films } from './data';
-import { FormField } from './components/FormField';
 import {
-  BrowserRouter,
+  HashRouter,
   Switch,
   Route,
 } from 'react-router-dom';
+import FilmsList from './components/FilmsList/FilmsList';
+import { NewFilm } from './components/NewFilm';
+import { FormField } from './components/FormField';
 import { FilmDetails } from './components/FilmDetails';
+import { store, addNewFilm } from './store';
 
-const API_URL = 'http://www.omdbapi.com/?apikey=2f4a38c9&t=';
+const API_URL = 'https://www.omdbapi.com/?apikey=2f4a38c9&t=';
 
 export class App extends Component {
   state = {
-    filmsList: films,
     searchWord: '',
   };
 
+  unsubscribe = null;
+
   componentDidMount() {
     this.searchFilm('spider');
+    this.unsubscribe = store
+      .subscribe(() => this.setState());
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
   }
 
   handleAddFilm = (newFilm) => {
-    this.setState(prevState => ({
-      filmsList: [
-        ...prevState.filmsList,
-        {
-          id: prevState.filmsList[prevState.filmsList.length - 1].id + 1,
-          ...newFilm,
-        },
-      ],
+    const { films } = store.getState();
+    store.dispatch(addNewFilm({
+      ...newFilm,
+      id: films[films.length - 1].id + 1,
```

Comment:
would be great to generate unique id, otherwise it will be the same for the same films you added to the list